src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -564,27 +565,20 @@ public function getEditLink($id)
     private function getAdjacentRecordID($offset)
     {
         $gridField = $this->getGridField();
+        /** @var DataList $list */
         $list = $gridField->getManipulatedList();
-        $state = $gridField->getState(false);
-        $gridStateStr = $this->getStateManager()->getStateFromRequest($this->gridField, $this->getRequest());
-        if (!empty($gridStateStr)) {
-            $state->setValue($gridStateStr);
-        }
-        $data = $state->getData();
-        $paginator = $data->getData('GridFieldPaginator');
-        if (!$paginator) {
-            return false;
-        }
-
-        $currentPage = $paginator->getData('currentPage');
-        $itemsPerPage = $paginator->getData('itemsPerPage');
 
-        $limit = $itemsPerPage + 2;
-        $limitOffset = max(0, $itemsPerPage * ($currentPage-1) -1);
-
-        $map = $list->limit($limit, $limitOffset)->column('ID');
+        $table = DataObject::getSchema()->tableForField($this->record, 'ID');
+        if ($offset > 0) {
+            $list = $list->where(["\"$table\".\"ID\" >= ?" => $this->record->ID])
+                ->sort(['ID' => 'ASC']);
+        } else {
+            $list = $list->where(["\"$table\".\"ID\" <= ?"  => $this->record->ID])
+                ->sort(['ID' => 'DESC']);
```

Comment:
Doesn’t this change the sorting completely? Any sort order applied to the gridfield (or a `default_sort`) will be ignored, and the records will be sorted by ID. Is that intended?

Reply:
> Doesn’t this change the sorting completely? Any sort order applied to the gridfield (or a `default_sort`) will be ignored, and the records will be sorted by ID. Is that intended?

Hmm, I'm pretty sure `$list->getManipulatedList()` already applies all the filters.